homeassistant/components/fritzbox/config_flow.py

```diff
@@ -161,6 +161,7 @@ async def async_step_confirm(
             result = await self.hass.async_add_executor_job(self._try_connect)
 
             if result == RESULT_SUCCESS:
+                assert isinstance(self._name, str)
```

Comment:
It's usually better to check `None`:
```suggestion
                assert self._name is not None
```